adapta/storage/distributed_object_store/v3/datastax_astra/_model_mappers.py

```diff
@@ -275,6 +277,143 @@ def _get_original_types(
         return {field.name: field.type for field in fields(self._data_model) if not subset or field.name in subset}
 
 
+class PanderaPolarsMapper(CassandraModelMapper):
+    """Maps Pandera Polars data models to Cassandra models."""
+
+    def __init__(
+        self,
+        data_model: Type[pandera.polars.DataFrameModel],
+        keyspace: Optional[str] = None,
+        table_name: Optional[str] = None,
+        primary_keys: Optional[List[str]] = None,
+        partition_keys: Optional[List[str]] = None,
+        custom_indexes: Optional[List[str]] = None,
+    ):
+        super().__init__(
+            data_model=data_model,
+            keyspace=keyspace,
+            table_name=table_name,
+            primary_keys=primary_keys,
+            partition_keys=partition_keys,
+            custom_indexes=custom_indexes,
+        )
+        self._data_model_schema = data_model.to_schema()
+
+    def _map_to_column(
+        self,
+        type_to_map: Type,
+    ) -> typing.Union[
+        typing.Tuple[Type[columns.List],],
+        typing.Tuple[Type[columns.Map],],
+        typing.Tuple[Type[Column],],
+        typing.Tuple[Type[Column], Type[Column]],
+        typing.Tuple[Type[Column], Type[Column], Type[Column]],
+        typing.Tuple[Type[columns.List], columns.Map],
+    ]:
+        try:
+            # Attempt to map with native python types
+            return super()._map_to_column(type_to_map)
```

Comment:
if this returns, what does the code below do? :/

Reply:
The call is wrapped in a `try/except` block. Here's how it works:

1. **Mapping Attempt**: We first attempt to map the type using the `_map_to_column` function from `super()`. This function maps built-in Python types to Cassandra columns, and (if I recall correctly) it was originally implemented by you.

2. **Fallback for Polars Types**: If `super()._map_to_column` raises a `TypeError` (indicating no match), we then attempt to match the type using Polars types.

This custom mapping is necessary because Pandera's datamodels use Polars types instead of native Python types. I call `super()._map_to_column` as part of a workaround. Some Cassandra types, like mappings, aren't natively supported by Polars and are typed as `polars.Object`. By allowing the specification of the native Python type through metadata, this code can effectively handle these cases.

For reference, in the tests, you can find [an example of the type being specified in metadata](https://github.com/SneaksAndData/adapta/blob/0b35f40ef98aa1ac4930507fd6f007949e0d5678/tests/astra/test_cassandra_model_mapper.py#L61). In the mapper class, this type is [used here](https://github.com/SneaksAndData/adapta/blob/0b35f40ef98aa1ac4930507fd6f007949e0d5678/adapta/storage/distributed_object_store/v3/datastax_astra/_model_mappers.py#L360).

Any suggestions on how to do it cleaner?

Reply:
As discussed offline, the try/except has been removed by a simple refactor.

For potential future reference, there is an open feature request in Polars for supporting maps. If/when this is supported, we can possibly simplify the type handling for pandera.polars DataFrameModels.
https://github.com/pola-rs/polars/issues/8385